cmd/datamon/cmd/bundle_diff.go

```diff
@@ -0,0 +1,110 @@
+// Copyright © 2018 One Concern
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"text/template"
+
+	"github.com/oneconcern/datamon/pkg/core"
+	"github.com/oneconcern/datamon/pkg/storage/gcs"
+	"github.com/oneconcern/datamon/pkg/storage/localfs"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+var bundleDiffCmd = &cobra.Command{
+	Use:   "diff",
+	Short: "Diff a downloaded bundle with a remote bundle.",
+	Long: "Diff a downloaded bundle with a remote bundle.  " +
+		"--destination is a location previously passed to the `bundle download` command.",
+	Run: func(cmd *cobra.Command, args []string) {
+		const listLineTemplateString = `{{.Type}} , {{.Name}} , {{with .Additional}}{{.Size}} , {{.Hash}}{{end}} , {{with .Existing}}{{.Size}} , {{.Hash}}{{end}}`
+		listLineTemplate := template.Must(template.New("list line").Parse(listLineTemplateString))
+
+		sourceStore, err := gcs.New(params.repo.MetadataBucket, config.Credential)
+		if err != nil {
+			logFatalln(err)
+		}
+		blobStore, err := gcs.New(params.repo.BlobBucket, config.Credential)
+		if err != nil {
+			logFatalln(err)
+		}
+		path, err := sanitizePath(params.bundle.DataPath)
+		if err != nil {
+			logFatalln("Failed path validation: " + err.Error())
+		}
+		fs := afero.NewBasePathFs(afero.NewOsFs(), path+"/")
+		destinationStore := localfs.New(fs)
+
+		err = setLatestOrLabelledBundle(sourceStore)
+		if err != nil {
+			logFatalln(err)
+		}
+
+		localBundle := core.New(core.NewBDescriptor(),
+			core.ConsumableStore(destinationStore),
+		)
+		remoteBundle := core.New(core.NewBDescriptor(),
+			core.Repo(params.repo.RepoName),
+			core.MetaStore(sourceStore),
+			core.BlobStore(blobStore),
+			core.BundleID(params.bundle.ID),
+		)
+
+		if err = core.PublishMetadata(context.Background(), localBundle); err != nil {
```

Comment:
these `PublishMetadata` calls are probably better included in `core.Diff` for consistency